experimental/cuda2/event_semaphore.c

```diff
@@ -255,10 +286,26 @@ static iree_status_t iree_hal_cuda2_semaphore_wait(
 
   iree_time_t deadline_ns = iree_timeout_as_deadline_ns(timeout);
 
+  // Slow path: try to see if we can have a device CUevent to wait on. This
+  // should happen outside of the lock given that acquiring has its own internal
+  // locks. This is faster than waiting on a host timepoint.
+  iree_hal_cuda2_event_t* wait_event = NULL;
+  iree_status_t status = iree_hal_cuda2_semaphore_acquire_event_host_wait(
+      semaphore, value, &wait_event);
+  if (iree_status_is_ok(status)) {
+    IREE_CUDA_RETURN_AND_END_ZONE_IF_ERROR(
+        z0, semaphore->symbols,
+        cuEventSynchronize(iree_hal_cuda2_event_handle(wait_event)),
+        "cuEventSynchronize");
+    iree_hal_cuda2_event_release(wait_event);
+    IREE_TRACE_ZONE_END(z0);
+    return status;
```

Comment:
This logic is a little confusing to scan... normally
```c
if (...status)) {
  ...
  return status;
}
```
code is used to handle errors, but here it is used if the status is ok?

The way this is set up now, all non-ok status values fall through to the next block. Is that okay? (Can it return device lost? What error is expected if no event is found?)

Reply:
There are only two kinds of results. I guess I don't really need the `iree_status_t` here. So changed to a `bool` value to make it simpler.